### PR TITLE
[SOW MS3] Enable gesvdj API to be called

### PR DESCRIPTION
**Pull request by alugorey**
gesvdj re-enabled after fix came in from hipSOLVER: https://ontrack-internal.amd.com/browse/SWDEV-345921?focusedCommentId=9554282&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel

### CODE REVIEW

File: aten/src/ATen/native/cuda/linalg/BatchLinearAlgebraLib.cpp [hunk 1]
```diff
@@ -956,13 +950,8 @@ void svd_cusolver(const Tensor& A,
...
   static const char* check_svd_doc = "Check doc at https://pytorch.org/docs/stable/generated/torch.linalg.svd.html";
 
-#ifdef ROCM_VERSION
-  // ROCM only has gesvd implemented
-  const auto driver_v = c10::string_view("gesvd");
-#else
   // The default heuristic is to use gesvdj driver
-  const auto driver_v = driver.value_or("gesvdj");
-#endif
+  const auto driver_v = c10::string_view("gesvdj");
```
Comment (jithunnair-amd) [inline aten/src/ATen/native/cuda/linalg/BatchLinearAlgebraLib.cpp:954]: @alugorey Let's retain the CUDA code (`const auto driver_v = driver.value_or("gesvdj");`) as a reference.

**Comment by jithunnair-amd** [comment]
From http://rocmhead.amd.com:8080/job/pytorch/job/pytorch-test-1/264/consoleText:
```
Aug 23 22:36:17 Running test_linalg ... [2022-08-23 22:36:17.657160]
...
Aug 23 22:41:48 ----------------------------------------------------------------------
Aug 23 22:41:48 Ran 741 tests in 329.088s
Aug 23 22:41:48 
Aug 23 22:41:48 OK (skipped=68)
```

File: aten/src/ATen/native/cuda/linalg/BatchLinearAlgebraLib.cpp (hunk 1 above)
Comment (alugorey) [reply aten/src/ATen/native/cuda/linalg/BatchLinearAlgebraLib.cpp:955]: @jithunnair-amd done!

File: aten/src/ATen/native/cuda/linalg/BatchLinearAlgebraLib.cpp [hunk 2]
```diff
@@ -709,12 +709,9 @@ inline static void apply_svd_cusolver_gesvdj(const Tensor& A, const Tensor& U, c
   gesvdjInfo_t gesvdj_params;
   TORCH_CUSOLVER_CHECK(cusolverDnCreateGesvdjInfo(&gesvdj_params));
 
-  // ROCM does not support those functions
-#ifdef CUDART_VERSION
   // Todo: expose the following two parameters to users
   TORCH_CUSOLVER_CHECK(cusolverDnXgesvdjSetTolerance(gesvdj_params, std::numeric_limits<scalar_t>::epsilon()));
```
Comment (jithunnair-amd) [inline aten/src/ATen/native/cuda/linalg/BatchLinearAlgebraLib.cpp:713]: @alugorey It seems there's either a duplication or a typo in the hipify mappings for these functions: https://github.com/alugorey/pytorch/blob/3cad1fb057254136387970ce0e74d354b397ae81/torch/utils/hipify/cuda_to_hip_mappings.py#L8208. Can you please confirm and rectify the mappings as part of this PR?

**Pull request merged**